### ISSUE TITLE
Black is not removing unnecessary + signs in concatenated strings

### ISSUE THREAD

**Issue by tmotyl** [psf/black#2268]
**Describe the bug** A clear and concise description of what the bug is.

```
second_string = ("def"
    + "ddd" + "ddd")
```
is formated to 
```
second_string = "def" + "ddd" + "ddd"

```
where I would expect:
```
second_string = "defdddddd"
```

**To Reproduce** Steps to reproduce the behavior:

https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ADPAHpdAD2IimZxl1N_Wl9XM4Hukmv0lHvH8ZV8H_6vvOURMPls_WzMNcku5-VbAQaYsBSET4vbhPFAzFY5DcRsFuGPJMGD-DLKFHBbxUtuKRJa5J0G0_Yx5DOsXpPSCHuM1nMZEs7HyKRNJp5Z55EyifvKOiVnNdKMFmwsCkKQAAAAd1B6wN16dJoAAZYB0AEAAJ3bnBKxxGf7AgAAAAAEWVo=

**Expected behavior** A clear and concise description of what you expected to happen.

**Environment (please complete the following information):**

- Version: main

**Does this bug also happen on main?** To answer this, you have two options:
yes

**Comment by felix-hilden** [comment]
Moreover, as presented in your example, strings on the same line originally are not concatenated either.

```python
long_string = "abc" + "def"
```

Now this functionality is not to be expected, but... given that the feature is [advertised](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings) (and I was the one that wrote the damn paragraph in #2106, *sigh* 😄), and I can't make it happen with the `--experimental-string-processing` flag, this should be addressed in one of two ways:
- Implement the functionality (I'd say a reasonable suggestion)
- Clarify the documentation to say that only parenthesised strings are concatenated

[Playground](https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ACTAGBdAD2IimZxl1N_WmOFITz2LPkWOwa8iF9PPG9krDfaHViyS-iCAZ5QSGopMjNG36veOyrdbGZiyHA4xzN_QlWDZSCBadnMyy9_wYCVkvcj7jDVw8liu5kr9MlqYV14yyblGABHpqHV2jV94wABfJQBAAAA0v-PI7HEZ_sCAAAAAARZWg==), but this works locally as expected with `--experimental-string-processing`. What do you guys think?

**Comment by JelleZijlstra** [comment]
It's fine to do for string literals. The compiler does this already anyway.
```
In [196]: dis.dis('"a" + "b"')
  1           0 LOAD_CONST               2 ('ab')
              2 RETURN_VALUE
```

**Comment by ItsCinnabar** [comment]
> Maybe the idea is cool, but it's not something that should be part of Black. AFAIK Black does do not touch the behavior and here we remove the concatenation operation.

Agreed. I personally don't think this is something black should do.

**Comment by felix-hilden** [comment]
Just to note, this is the polar opposite of #2553.

**Comment by yilei** [comment]
Hmm, this isn't necessarily the opposite of #2553.

There are a few options that they can co-exist:
1. We could restrict this to cases where it's within the line length.
2. Or, when processing strings, we merge explicitly or implicitly concatenated strings first, then when performing the split (because it exceeds line length), we use explicit str concatenation.

Either case, the initial example could still be formatted to `second_string = "defdddddd"`.
